lib/ring.js

```diff
@@ -145,4 +146,39 @@ HashRing.prototype.lookup = function lookup(str) {
     return res;
 };
 
+// find (up to) N unique successor nodes (aka the 'preference list') for the given key
+HashRing.prototype.lookupN = function lookupN(str, n) {
+    // can't return more than the number of servers
+    var serverCount = this.getServerCount();
+    if (n > serverCount) {
+        n = serverCount;
+    }
+
+    var resultArray = [];
+    var resultSet = {}; // for fast dedup
+    var hash = this.hashFunc(str);
+    var iter = this.rbtree.upperBound(hash);
+    var wrappedAround = false;
+    while (resultArray.length < n) {
+        var res = iter.str();
+        if (res === null) {
+            // must not wrap around more than once
+            if (wrappedAround) {
```

Comment:
Instead of wrappedAround as a bool, how about as the original `iter.res()`. This should prevent unnecessary duplication of lookups, yea?